src/server/frontend_wayland/text_input_v1.cpp

```diff
@@ -427,16 +430,13 @@ void TextInputV1::commit_state(uint32_t client_serial)
     if (pending_state && current_surface)
     {
         auto const hub_serial = ctx->text_input_hub->set_handler_state(handler, on_new_input_field, *pending_state);
+        pending_state_used = true;
```

Comment:
It seems wrong not to reset `pending_state` value after using it in `commit_state()`. Vis:

    pending_state.emplace();